spring-test/src/main/kotlin/org/springframework/test/web/servlet/MockMvcPerformDsl.kt

```diff
@@ -51,6 +52,10 @@ open class MockMvcPerformDsl(private val requestBuilder: MockHttpServletRequestB
 		MockMvcResultMatchersDsl(resultMatchers).dsl()
 	}
 
+	fun <T : MockMvcResultMatchersDsl> expectCustom(matcher: (MutableList<ResultMatcher>) -> T, dsl: T.() -> Unit) {
+		matcher(resultMatchers).dsl()
+	}
```

Comment:
I had to add an `expectCustom` to be able to take advantage of the extending the matchers